.changeset/kind-pets-film.md

```diff
@@ -0,0 +1,9 @@
+---
+'@openfn/language-mssql': patch
+---
+
+Optimize query handling by:
+
+- Revert to using `tedious.Request's callback` signature
+- Remove `rowCollectionOnRequestCompletion` option from configuration
```

Comment:
Technically this is a major change. But because I don't think anyone will have used this - and to do so would surely break the adaptor - I'm happy to call it a bug fix and leave it as a patch